### ISSUE TITLE
magit keybindings for interactive rebase

### ISSUE THREAD

**Issue by justbur** [syl20bnr/spacemacs#2174]
Is there some way to change the key binding hints at the bottom since they're wrong? They say "k" for delete commit or something like that for example. But "k" is now move one line up. Also, I couldn't figure out where "k" got remapped to.

**Comment by syl20bnr** [comment]
They are remapped via the macro `spacemacs/evilify-map` which automatically set wrappers on the next available bindings like `k` --> ` K`. The wrappers takes care of emacs and evilify state (emacs state key bindings should not be altered).

**Comment by justbur** [comment]
@syl20bnr Can you try it? "K" is not mapped for me, and there's no emacs state to go into ("C-z" does nothing). It seems like a special "magit select" buffer. I don't have time right now to dig into what's going on, but it doesn't seem to be working. 

I'm on develop with emacs25.

**Comment by justbur** [comment]
I got in through `SPC g s` then `r e`

**Comment by syl20bnr** [comment]
This is because I overload it when passing additonal bindings :-(
https://github.com/syl20bnr/spacemacs/blob/develop/contrib/!source-control/git/packages.el#L191-L201

Weird that it does nothing for you though.

We should put the additional bindings  on `C-k` and `C-j` then.

**Comment by justbur** [comment]
Some more information. I didn't notice these messages before, although they may not be relevant to this exactly. These are recorded right after `SPC g s`

```
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-mode-map':
   - `magit-git-command' originally mapped on `:''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-mode-map':
   - `magit-diff-show-or-scroll-up' originally mapped on `SPC''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-blame-mode-map':
   - `magit-diff-show-or-scroll-up' originally mapped on `SPC''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-diff-mode-map':
   - `scroll-up' originally mapped on `SPC''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-log-read-revs-map':
   - `self-insert-command' originally mapped on `SPC''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `git-rebase-mode-map':
   - `scroll-up-command' originally mapped on `SPC''
(Spacemacs) Warning: Auto-evilication could not remap these functions in map `git-rebase-mode-map':
   - `scroll-up-command' originally mapped on `SPC''
```

**Comment by syl20bnr** [comment]
This is not related. I'm aware of these warnings, I have to find good
replacements keys for SPC, : and /, for letters it was easy. Any ideas is
welcome.

Le jeudi 2 juillet 2015, Justin Burkett <notifications@github.com> a écrit :

> Some more information. I didn't notice these messages before, although
> they may not be relevant to this exactly. These are recorded right after SPC
> g s
>
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-mode-map':
>    - `magit-git-command' originally mapped on `:''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-mode-map':
>    - `magit-diff-show-or-scroll-up' originally mapped on `SPC''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-blame-mode-map':
>    - `magit-diff-show-or-scroll-up' originally mapped on `SPC''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-diff-mode-map':
>    - `scroll-up' originally mapped on `SPC''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `magit-log-read-revs-map':
>    - `self-insert-command' originally mapped on `SPC''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `git-rebase-mode-map':
>    - `scroll-up-command' originally mapped on `SPC''
> (Spacemacs) Warning: Auto-evilication could not remap these functions in map `git-rebase-mode-map':
>    - `scroll-up-command' originally mapped on `SPC''
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/issues/2174#issuecomment-118046804>
> .
>


-- 
-syl20bnr-

**Comment by justbur** [comment]
I don't mind too much where the keybindings go (although I wish "k" for kill could be remapped to "d"). I was more interested in doing something with the comment section, so that it's not completely wrong. Unfortunately it seems to be hardcoded: https://github.com/magit/magit/blob/fb3e25dd776d83d5cb459a183bc128c652375d9e/lisp/git-rebase.el#L341-L367

**Comment by syl20bnr** [comment]
You know like me that there is no such thing has "hardcoded" in elisp ;-)
We can advice it.

If d is free them Insee no problem to tweak the map to include it. We can
do it with :bindings of the macro.

Le jeudi 2 juillet 2015, Justin Burkett <notifications@github.com> a écrit :

> I don't mind too much where the keybindings go (although I wish "k" for
> kill could be remapped to "d"). I was more interested in doing something
> with the comment section, so that it's not completely wrong. Unfortunately
> it seems to be hardcoded:
> https://github.com/magit/magit/blob/fb3e25dd776d83d5cb459a183bc128c652375d9e/lisp/git-rebase.el#L341-L367
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/issues/2174#issuecomment-118049613>
> .
>


-- 
-syl20bnr-

**Comment by justbur** [comment]
Yes, of course :-) 

Any ideas on `C-z` not working? It would be nice to be able to quickly revert to the default behavior if you get lost in the evil keybindings. Or is it not supposed to work in this case

**Comment by syl20bnr** [comment]
I guess evil has a dedicated variable for the temporary state, we can support it in the wrappers.

**Comment by justbur** [comment]
I'm not sure a temporary state is what I'm thinking about. I can't explain why, but sometimes the vim-like keybindings are more natural to me but other times the emacs-like ones are. Could we maybe have a variable that prevents certain maps from being evilified? I appreciate the effort you put into making a consistent interface, but I still find myself getting lost and confused (maybe that's my problem and not yours).

**Comment by justbur** [comment]
Never mind. That's probably going to make it more complicated. 

There's just so many different states now that it's hard to keep track of what's going on sometimes.

**Comment by syl20bnr** [comment]
@justbur you can remove the magit modes from `evil-evilified-state-modes` and add them to `evil-emacs-state-modes` so you'll be in emacs state and then you'll get stock emacs bindings.
Note that this behavior has no changed, the evilified magit buffer were already in 102 and prior versions.

**Comment by justbur** [comment]
Oh, I didn't know about those variables. Thanks. 

Sorry, I wasn't saying something had changed. I was just reporting that the message in the rebase buffer was wrong about the keybindings and that I was having trouble figuring out the right ones. Thanks for your help

**Comment by justbur** [comment]
Resolved with #3578
